### PR TITLE
Install sccache from source

### PR DESCRIPTION
**Pull request by malfet**
Build `sccache` from https://github.com/pytorch/sccache

Also, update sccache wrappers not to call sccache from sccache

### CODE REVIEW

**Comment by malfet** [comment]
@ezyang , is there any reason trying to build sccache as static binary if we are building it on a per-distro basis?

File: .circleci/docker/common/install_cache.sh [hunk 1]
```diff
@@ -11,7 +33,17 @@ export PATH="/opt/cache/bin:$PATH"
...
+  ID=$(grep -oP '(?<=^ID=).+' /etc/os-release | tr -d '"')
+  case "$ID" in
+    ubuntu)
+      install_ubuntu
+      ;;
+    *)
+      install_binary
+      ;;
+  esac
+  install_binary
```
Comment (walterddr) [inline .circleci/docker/common/install_cache.sh:45]: why running install_binary twice? typo?

Comment (malfet) [reply .circleci/docker/common/install_cache.sh]: Fixed

**Comment by ezyang** [comment]
> @ezyang , is there any reason trying to build sccache as static binary if we are building it on a per-distro basis?

No, I don't think so. Static binary is just for extra portability, which apparently doesn't matter here.

File: .circleci/docker/common/install_cache.sh [hunk 2]
```diff
@@ -11,7 +33,17 @@ export PATH="/opt/cache/bin:$PATH"
...
+  apt-get install cargo pkg-config libssl-dev
```
[the remark this answers is not in the document]
Comment (malfet) [reply .circleci/docker/common/install_cache.sh]: I think I've deleted this line already, don't I?

File: .circleci/docker/common/install_cache.sh [hunk 3]
```diff
@@ -38,8 +68,8 @@ if [ -n "$CUDA_VERSION" ]; then
   # where CUDA is installed.  Instead, we install an nvcc symlink outside
   # of the PATH, and set CUDA_NVCC_EXECUTABLE so that we make use of it.
 
-  printf "#!/bin/sh\nexec sccache $(which nvcc) \"\$@\"" > /opt/cache/lib/nvcc
-  chmod a+x /opt/cache/lib/nvcc
+  write_sscache_stub nvcc
```
Comment (janeyx99) [inline .circleci/docker/common/install_cache.sh:71]: write_sccache_stub typo

**Review by janeyx99 — APPROVED** [review]

**Pull request closed without merging**